aspnetcore/blazor/tutorials/movie-database-app/part-2.md

```diff
@@ -0,0 +1,500 @@
+```
+
+*CRUD* is an acronym for Create, Read, Update, and Delete. The `blazor` generator with the `CRUD` template produces create, edit, delete, details, and index components for the app.
+
+The following table details the ASP.NET Core code generator options used in the preceding command:
+
+* `-dbProvider`: Database provider to use. Options include `sqlserver` (default), `sqlite`, `cosmos`, `postgres`.
+* `-dc`: The <xref:Microsoft.EntityFrameworkCore.DbContext> class to use, including the namespace (`BlazorWebAppMovies.Data`).
+* `-m`: The name of the model.
+* `-outDir`: The output directory for the generated components. A folder is created from the model name in the output directory to hold the components (for example, `MoviePages` in this case).
+
+:::zone-end
+
+The `appsettings.json` file is updated with the connection string used to connect to a local database. In the following example, the `{CONNECTION STRING}` is the connection string automatically generated by the scaffolder:
+
+```json
+"ConnectionStrings": {
+  "BlazorWebAppMoviesContext": "{CONNECTION STRING}"
+}
+```
+
+## Files created and updated by scaffolding
+
+The scaffolding process creates the following component files and movie database context class file:
+
+* `Components/Pages/MoviePages`
+  * `Create.razor`: Creates new movie entities.
+  * `Delete.razor`: Deletes a movie entity.
+  * `Details.razor`: Shows movie entity details.
+  * `Edit.razor`: Updates a movie entity.
+  * `Index.razor`: Lists movie entities (records) in the database.
+* `Data/BlazorWebAppMoviesContext.cs`: Database context file (<xref:Microsoft.EntityFrameworkCore.DbContext>).
+
+The component files in the `MoviePages` folder are described in greater detail in the next part of this tutorial. The database context is described later in this article.
+
+ASP.NET Core is built with dependency injection, which is a software design pattern for achieving [Inversion of Control (IoC)](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#dependency-inversion) between classes and their dependencies. Services, such as the EF Core database context, are registered with dependency injection during application startup. These services are injected into Razor components for use by the components.
+
+The <xref:Microsoft.AspNetCore.Components.QuickGrid> component is a Razor component for efficiently displaying data in tabular form. The scaffolder places a <xref:Microsoft.AspNetCore.Components.QuickGrid> component in the `Index` component (`Components/Pages/Index.razor`) to display movie entities. Calling <xref:Microsoft.Extensions.DependencyInjection.EntityFrameworkAdapterServiceCollectionExtensions.AddQuickGridEntityFrameworkAdapter%2A> on the service collection adds an EF Core adapter for <xref:Microsoft.AspNetCore.Components.QuickGrid> to recognize EF Core-supplied <xref:System.Linq.IQueryable%601> instances and to resolve database queries asynchronously for efficiency.
```

Comment:
Each of these references to `QuickGrid` is rendering the full type name. Is that expected? 

![image](https://github.com/user-attachments/assets/ebf7f0d0-4be0-41ab-95e1-57ac618638b5)

Reply:
There's no `?displayProperty=fullName` on them, so they shouldn't appear fully qualified. I'll change these to use link formatting just to get us past this problem.

@v-alje ... `QuickGrid` API was added per https://github.com/dotnet/aspnetcore/issues/52466. Was there anything in particular that was done when it was added that would cause this? All I have in the article is ...

```
<xref:Microsoft.AspNetCore.Components.QuickGrid>
```

... i.e., I'm not applying `?displayProperty=fullName` to these, so the full name syntax is unexpected.

Reply:
Is it maybe because the type lives in a separate NuGet package?

Reply:
🤔 ... I think this is happening because that's a namespace name. The correct object name for a QG ***grid***  is a `QuickGrid<TGridItem>`.

Nevermind, @v-alje ... I think I've figured out what's going on with it. I'll go with link-style cross-links on the next commit, and that will take care of them.

Reply:
In order to "fix" it correctly, the link would be ...

```
<xref:Microsoft.AspNetCore.Components.QuickGrid.QuickGrid`1>
```

... but I think under the circumstances here that we're fine if the dev is taken to the namespace page at ...

https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.components.quickgrid

... which gives them a better overview of the entire API.

Also, setting the link text will avoid it showing up as `QuickGrid<TGridItem>` ... the `<TGridItem>` part is a bit overkill given what we're covering here in an introductory tutorial.